### PR TITLE
test: regression coverage for the #738 OpenRouter council seat fixes

### PR DESCRIPTION
**Pull request by nyldn**
## Description

Follow-up requested by @nyldn in [review of #739](https://github.com/nyldn/claude-octopus/pull/739#issuecomment-generated):

> **Gap: no tests.** `run_with_timeout` has 11 existing unit suites and `openrouter_execute` has 1, but none covers these three behaviours — so nothing stops any of them regressing. [...] I'd like a follow-up adding:
> 1. `kill`/`wait` on an already-reaped monitor does not abort `run_with_timeout` and the captured output survives;
> 2. `council_detect_providers` reports `openrouter` available on `OPENROUTER_API_KEY` alone, with no binary on PATH;
> 3. `openrouter_execute` honours `OCTOPUS_OPENROUTER_MODEL` over the task-type table.

This PR adds `tests/unit/test-issue-738-openrouter-council-seat.sh` with 8 tests covering all three, plus the fallback ordering (`OCTOPUS_OPENROUTER_MODEL` → `resolve_octopus_model` → task-type table) and the negative case (`OPENROUTER_API_KEY` unset → `missing`).

Each test was verified against both sides of the fix landed in #739:
- Temporarily swapped in the pre-#739 versions of `council.sh`, `heartbeat.sh`, `perplexity.sh` → 4 of the 8 tests fail (the ones covering the actual bugs), 4 pass (guard-behavior/negative-path tests that were already correct pre-fix).
- Restored the merged fix → all 8 pass.

This confirms the tests are not vacuous.

## Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactoring
- [x] Other (describe): test-only, no production code changed

## Checklist
- [x] Code passes `bash -n scripts/orchestrate.sh`
- [x] Shell scripts pass `bash -n` syntax check
- [x] Tests pass: new suite + all directly related existing suites (see Testing below)
- [ ] OpenClaw registry in sync — N/A, no tool/command surface changed
- [ ] New skills/commands registered — N/A
- [ ] Version bump — N/A, test-only change
- [ ] Documentation updated — N/A
- [ ] CHANGELOG.md updated — left for the release PR, per repo convention

## Testing

```bash
bash -n tests/unit/test-issue-738-openrouter-council-seat.sh

bash tests/unit/test-issue-738-openrouter-council-seat.sh   # 8 passed (new)
bash tests/unit/test-council-command.sh                     # 67 passed
bash tests/unit/test-heartbeat-timeout.sh                   # 12 passed
bash tests/unit/test-perplexity-issue-307.sh                # 6 passed
bash tests/unit/test-provider-health-probe.sh                # 12 passed
bash tests/unit/test-council-model-selection-fixes.sh        # 10 passed
```

## Related Issues
Follow-up to #738 / #739

---
_Generated by [Claude Code](https://claude.ai/code/session_01At8qoUSS8omaktiiou7wSA)_

## Summary by CodeRabbit

* **Tests**
  * Added regression coverage for heartbeat monitor cleanup and timeout handling, including preservation of fallback output.
  * Added tests for OpenRouter provider detection, API-key validation, and model selection precedence.
  * Verified behavior when provider credentials are missing and when models are selected through environment settings, resolvers, or task-type fallbacks.
  * Expanded coverage for configuration edge cases and provider selection reliability.

### CODE REVIEW

File: tests/unit/test-issue-738-openrouter-council-seat.sh [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    wait "$m" 2>/dev/null || true   # reaps the monitor, simulating it having already exited
+    kill "$m" 2>/dev/null || true   # now-guarded: kill on an already-reaped pid
+    wait "$m" 2>/dev/null || true   # now-guarded: wait on an already-reaped pid
+    echo "REACHED-END"
+')
+if [[ "$out" == "REACHED-END" ]]; then
+    test_pass
+else
+    test_fail "expected REACHED-END, got: '$out' (set -e aborted before reaching the end)"
+fi
```
[the remark this answers is not in the document]
Comment (nyldn) [reply tests/unit/test-issue-738-openrouter-council-seat.sh:56]: Fair point that this specific sub-test can't fail on its own — it carries its own `|| true` guards, so it passes on both sides of #739 by construction.

Leaving it as-is rather than rewriting it into a real `run_with_timeout` invocation: it's the maintainer's own reproduction from the #739 review thread, explicitly requested "lifted almost verbatim." Making it exercise the real `heartbeat.sh` monitor would need the wrapped command to survive past the monitor's full `sleep timeout_secs; SIGTERM; sleep 10; SIGKILL` cycle (~11s+ minimum, plus a SIGTERM-trapping stub), which is a lot of added complexity and wall-clock time for what's already covered elsewhere.

The regression detection for the actual code lives in the next test ("heartbeat.sh guards both the monitor kill and the monitor wait with `|| true`"), which I verified fails against pre-#739 `heartbeat.sh` and passes against the merged fix. Added a comment above the repro clarifying that split in 3053aa1's follow-up commit (df907147) so it doesn't read as a broken regression test on its own.


---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nyldn** [comment]
Merged via #764 (squash `ac23a06f`), which batched nine independent PRs into one CI cycle. GitHub does not auto-close sources when an integration branch is squash-merged, so closing manually rather than leaving it dangling.

Verified this PR's content is on `main` before closing — not assumed from the merge alone.

Context: branch protection has `strict: true`, so with a dozen PRs open each merge invalidated the rest at a full CI cycle each. A merge queue would be the right fix but is unavailable — this is a user-owned repository and merge queue is organization-only. All nine merged with **zero conflicts**, which confirms the serialization was pure overhead rather than real interaction.
